### ISSUE TITLE
Chapter 3: Fit before cross_val_predict()?

### ISSUE THREAD

**Issue by girishsg24** [ageron/handson-ml#112]
Hello I have a doubt regarding, cross_val_predict().
Here is the code that I am referring to from chapter 3

In [14]:
from sklearn.linear_model import SGDClassifier
sgd_clf = SGDClassifier(random_state=42)
sgd_clf.fit(X_train, y_train_5)
**Here the model is first fit using X_train and y_train_5**

In [20]:
from sklearn.model_selection import cross_val_predict
y_train_pred = cross_val_predict(sgd_clf, X_train, y_train_5, cv=3)
**Then the fit model is again used to get predicted values** 

From the definition of cross_val_predict I have understood that the given model is cloned for each cross validation set and predicts the values for the data in crossvalidation test set. It says that y_train_pred has clean predictions at the end where each model makes prediction for the data never seen before.

But the model passed as argument to the cross_val_pred has already seen the data and is fit to the X_train. Shouldn't we send an untrained SGDClassifier to the cross_val_predict. Could you please suggest, I am still a rookie trying to understand the process and feel free to correct my understanding if i am wrong.

**Comment by ageron** [comment]
Hi @girishsg24 ,
Indeed, the `cross_val_predict()` will clone the estimator we give it, and it will call its `fit()` method, which will just train the estimator from scratch, so it makes no difference whether the estimator we give it is already trained or not.  Some estimators have a `warm_start` parameter: if you set it to `True`, then calling `fit()` will *continue* training rather than start over from scratch.  Since I did not set `warm_start=True`, it really makes no difference whether the estimator was already trained or not.
Hope this helps,
Aurélien

**Comment by girishsg24** [comment]
@ageron Thanks for clarifying this.
